### PR TITLE
CVSL-2241 add dpr component library and dependencies

### PR DESCRIPTION
**Pull request by louisenorris**

### CODE REVIEW

File: server/views/layout.njk [hunk 1]
```diff
@@ -121,8 +121,21 @@
 
 {% block bodyEnd %}
   {# Run JavaScript at end of the <body>, to avoid blocking the initial render. #}
-  <script type="module" src="/assets/govukFrontendInit.js"></script>
+  <script type="module" src="/assets/ext/day.js"></script>
+  <script type="module" src="/assets/ext/dayjs/plugin/customParseFormat.js"></script>
+  <script type="module" src="/assets/ext/chart.js"></script>
+  <script type="module" src="/assets/ext/chartjs-datalabels.js"></script>
```
Comment (andrewrlee) [inline server/views/layout.njk:127]: Presumably we don't need to load all this on every page across the app? 
Is this just required on pages that show dashboards?

If this is in their documentation can you get them to update it?

Comment (louisenorris) [reply server/views/layout.njk:127]: The guidance specifies to add the scripts to the nunjucks layout but I'll double check this with them

File: assets/sass/application.sass [hunk 2]
```diff
@@ -5,6 +5,7 @@ $path: "/assets/images/"
 
 @import 'govuk/all'
 @import 'moj/all'
+@import "node_modules/@ministryofjustice/hmpps-digital-prison-reporting-frontend/dpr/all"
```
Comment (andrewrlee) [inline assets/sass/application.sass:8]: is it possible to add a load-path to the `bin/build-css.sh` so we don't need to reference the file structure here?

File: server/middleware/setUpStaticResources.ts [hunk 3]
```diff
@@ -37,6 +37,30 @@ export default function setUpStaticResources(): Router {
...
+  ;['/node_modules/dayjs/dayjs.min.js'].forEach(dir => {
+    router.use('/assets/ext/day.js', express.static(path.join(process.cwd(), dir), cacheControl))
+  })
+  ;['/node_modules/dayjs/plugin/customParseFormat.js'].forEach(dir => {
+    router.use(
+      '/assets/ext/dayjs/plugin/customParseFormat.js',
+      express.static(path.join(process.cwd(), dir), cacheControl),
+    )
+  })
+  ;['/node_modules/govuk-frontend/dist/govuk/govuk-frontend.min.js'].forEach(dir => {
```
Comment (andrewrlee) [inline server/middleware/setUpStaticResources.ts:61]: **issue:** we already have an entry for govuk-frontend above.

**suggestion:** We don't need to iterate over arrays of one item. 
It'd be better to iterate over all items or if that's not possible then just inline each call

Comment (louisenorris) [reply server/middleware/setUpStaticResources.ts]: Cool, have updated to no longer iterate where not needed and removed duplication of entry for govuk-frontend - sorry thats me thinking the second was different, introduced via the DPR guidance.

File: server/views/layout.njk [hunk 4]
```diff
@@ -121,8 +121,13 @@
 
 {% block bodyEnd %}
   {# Run JavaScript at end of the <body>, to avoid blocking the initial render. #}
+  <!-- Govuk Lib -->
   <script type="module" src="/assets/govukFrontendInit.js"></script>
+  <!-- MoJ Lib -->
   <script src="/assets/moj/all.js"></script>
-  <script src="/assets/mojFrontendInit.js"></script>
+  <script type="module" src="/assets/mojInit.mjs"></script>
```
Comment (andrewrlee) [inline server/views/layout.njk:128]: mmh.. what's the difference between moj frontend and moj all?

File: server/views/layout.njk [hunk 5]
```diff
@@ -121,8 +121,13 @@
...
+
+  <!-- DPR Lib -->
+  <script type="module" src="/assets/dpr/js/all.mjs"></script>
```
Comment (andrewrlee) [inline server/views/layout.njk:131]: I guess this only needs to included in pages that have reports too..

Comment (louisenorris) [reply server/views/layout.njk]: Removed this from layout - thank you

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**